app/src/main/java/com/example/app/ipc/OracleDriveServiceConnector.kt

```
@@ -6,83 +6,259 @@ import android.content.Intent
+                onServiceEvent?.invoke(EVENT_ERROR, "Service connection error: ${e.message}")
+            }
         }
 
         override fun onServiceDisconnected(name: ComponentName?) {
-            auraDriveService = null
-            _isServiceConnected.value = false
+            try {
+                serviceCallback?.let { callback ->
+                    auraDriveService?.unregisterCallback(callback)
+                }
+            } catch (e: RemoteException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)